### PR TITLE
Add Twitter/X adapter via twitterapi.io

### PR DESCRIPTION
**Pull request by user1303836**
## Summary

- Add `TwitterAdapter` that polls tweets from Twitter/X accounts using the twitterapi.io API (`GET /twitter/user/last_tweets`)
- Support subscribing to Twitter accounts via `/source add type:Twitter name:"..." url:https://x.com/username`
- Optional `summarize` parameter (default True) -- when False, posts bare tweet URLs for Discord's native embed
- Skips retweets; includes quoted tweet text in content for richer AI summarization

## Changes

- **New adapter**: `src/intelstream/adapters/twitter.py` -- `TwitterAdapter` with `X-API-Key` auth, date parsing, retweet filtering
- **Config**: New optional `TWITTER_API_KEY` env var in `config.py`
- **Model**: `TWITTER` added to `SourceType` enum
- **Pipeline**: Adapter registered in `ContentPipeline._create_adapters()`, guarded by API key presence
- **Discord commands**: Twitter URL parsing (twitter.com + x.com), command choice, API key guard
- **Content poster**: Twitter label for formatted messages
- **README**: Updated with Twitter in features, setup, usage examples, and source-specific behavior
- **Tests**: 20 new adapter tests, 7 new URL parsing/API key tests, 2 new pipeline registration tests

## Test plan

- [ ] Verify all existing tests still pass (`uv run pytest tests/`)
- [ ] Verify lint/format/type checks pass (`uv run ruff check .`, `uv run ruff format --check .`, `uv run mypy src/`)
- [ ] Manual test: `/source add type:Twitter name:"Test" url:https://x.com/username` with a valid `TWITTER_API_KEY`
- [ ] Manual test: `/source add type:Twitter ...` without API key configured shows "not available" error
- [ ] Manual test: Add with `summarize:False` to verify bare URL posting

### CODE REVIEW

File: src/intelstream/adapters/twitter.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                f"{TWITTER_API_BASE}/twitter/user/last_tweets",
+                headers=headers,
+                params=params,
+            )
+        else:
+            async with httpx.AsyncClient() as client:
+                response = await client.get(
+                    f"{TWITTER_API_BASE}/twitter/user/last_tweets",
+                    headers=headers,
+                    params=params,
```
Comment (user1303836) [inline src/intelstream/adapters/twitter.py]: The fallback client path creates a new `httpx.AsyncClient()` per call with no timeout configured. The shared `self._client` from the pipeline has `timeout=self._settings.http_timeout_seconds` set during initialization.

If the adapter is ever used without a shared client (unlikely in production, but the code path exists), requests to a slow/unresponsive API could hang indefinitely. Consider adding a default timeout:
```python
async with httpx.AsyncClient(timeout=30.0) as client:
```

**Review by user1303836 — COMMENTED** [review]
## Final Review

All 3 actionable items from the previous review have been properly addressed in commit 2092aa4.

### One gap: `src/intelstream/services/summarizer.py:126-132`

The `_build_prompt` content_type mapping is missing a `"twitter"` entry:

```python
content_type = {
    "substack": "newsletter article",
    "youtube": "video transcript",
    "rss": "blog post",
    "web": "article",
    "arxiv": "research paper abstract",
}.get(source_type, "article")  # "twitter" falls through to default "article"
```

When tweets go through summarization, the prompt reads "Summarize the following **article** from..." instead of "tweet" or "social media post". This won't break anything, but Claude will apply article-length formatting (thesis + key arguments structure) to a 280-character tweet. Adding `"twitter": "tweet"` would produce better-tuned output.

### Everything else looks good

- **Documentation**: README additions are thorough and accurate (prerequisites, env vars, usage examples including `summarize:False`, architecture tree).
- **No breaking changes**: Twitter is fully opt-in behind `twitter_api_key`. All existing test mocks updated. Optional field defaults to `None`.
- **Follows codebase patterns**: Adapter pattern, pipeline registration gating, source management URL parsing, and API key guards all match existing conventions.
- **Test coverage**: 20 adapter tests, 7 source management tests (including path traversal edge case), 2 pipeline registration tests.
- **Security**: Username regex validation prevents injection. API key masked in `__repr__`.
